src/scripts/extensions/authenticationHelper.ts

```diff
@@ -67,18 +67,23 @@ export class AuthenticationHelper {
 			let getInfoEvent: Log.Event.PromiseEvent = new Log.Event.PromiseEvent(Log.Event.Label.GetExistingUserInformation);
 			getInfoEvent.setCustomProperty(Log.PropertyName.Custom.UserInformationStored, !!storedUserInformation);
 			this.clipperData.getFreshValue(ClipperStorageKeys.userInformation, getUserInformationFunction, updateInterval).then((response: TimeStampedData) => {
-				getInfoEvent.setCustomProperty(Log.PropertyName.Custom.FreshUserInfoAvailable, !!response);
+				let isValidUser = this.isValidUserInformation(response.data);
+				getInfoEvent.setCustomProperty(Log.PropertyName.Custom.FreshUserInfoAvailable, isValidUser);
 
-				if (response) {
-					this.user.set({ user: response.data, lastUpdated: response.lastUpdated, updateReason: updateReason });
+				let writeableCookies = this.isThirdPartyCookiesEnabled(response.data);
+				getInfoEvent.setCustomProperty(Log.PropertyName.Custom.WriteableCookies, writeableCookies);
+
+				if (isValidUser) {
+					this.user.set({ user: response.data, lastUpdated: response.lastUpdated, updateReason: updateReason, writeableCookies: writeableCookies });
 				} else {
-					this.user.set({ updateReason: updateReason });
+					this.user.set({ updateReason: updateReason, writeableCookies: writeableCookies });
 				}
 
 				resolve(this.user.get());
 			}, (error: OneNoteApi.GenericError) => {
 				getInfoEvent.setStatus(Log.Status.Failed);
 				getInfoEvent.setFailureInfo(error);
+
 				this.user.set({ updateReason: updateReason });
 				resolve(this.user.get());
```

Comment:
On a side note: This line repeats itself in the good and bad case, so we can consolidate it into the final then() block.